tests/Client/Response/ListGroupsResponseTest.php

```diff
@@ -21,20 +21,29 @@ final class ListGroupsResponseTest extends TestCase
 {
     public function testCanBeConstructed(): ListGroupsResponse
     {
-        $successfulCurlResponse = '{"groups":[{"group":"111111111111111111111111111111","name":"Group1"},{"group":"222222222222222222222222222222","name":"group2"},{"group":"333333333333333333333333333333","name":"Group 3"}],"status":1,"request":"aaaaaaaa-1111-bbbb-2222-cccccccccccc"}';
```

Comment:
I think it is better to use pure json e.g. mocking real response.

Reply:
Can you explain why? IMHO it is more a) more readable + b) easier to maintain and c) easier to copy&paste modify.

The only thing, where I would use real JSON string is, if I want to test how my code performs if the JSON is invalid in case of the syntax like `{"foo": null}}}}` (note the trailing brackets)

Reply:
The reason is very simple. Pushover API returns JSON. And I want to provide this JSON exactly as it is returned by the API into tests. I would like to avoid this "man-in-the-middle" conversion from PHP array into JSON. Whatever we get from the API should be directly supplied to the tests.

Reply:
Done